apps/mobile/src/modules/Swap/index.tsx

```diff
@@ -0,0 +1,365 @@
+import {parseUnits} from '@ethersproject/units';
+import {Feather} from '@expo/vector-icons';
+import React, {useCallback, useEffect, useState} from 'react';
+import {Image, ScrollView, Text, TextInput, TouchableOpacity, View, ViewStyle} from 'react-native';
+import {parseEther} from 'viem';
+import {useSendTransaction, useWriteContract} from 'wagmi';
+
+import {useSwapModal} from '../../context/SwapModalProvider';
+import {useEvmWallet} from '../../context/WalletModalEvmProvider';
+import {useStyles, useTheme} from '../../hooks';
+import {
+  useAvnuExecuteSwap,
+  useAvnuSwapCalldata,
+  useGetAvnuSwapQuoteDetails,
+  useGetEvmTokens,
+} from '../../starknet/evm/hooks';
+import styleSheet from './styles';
+import TokenSelectModal from './TokenSelection';
+import {formatToUSD} from './util';
+interface Token {
+  symbol: string;
+  l2_token_address: string;
+  decimals: number;
+  logo_url: string;
+}
+
+export default function TokenSwapView({showHeader = false}: {showHeader?: boolean}) {
+  const {showSwap} = useSwapModal();
+  const {sendTransaction} = useSendTransaction();
```

Comment:
AVNU is a Starknet DEX, we have hook for Starknet already and modal to connect Argent/Bravoos account